src/ImageSharp/Formats/AnimationUtilities.cs

```diff
@@ -0,0 +1,190 @@
+                }
+            }
+
+            for (i = remaining; i > 0; i--)
+            {
+                x = (uint)(length - i);
+
+                Vector4 p = Unsafe.Add(ref Unsafe.As<Vector256<float>, Vector4>(ref previousBase), x);
+                Vector4 c = Unsafe.Add(ref Unsafe.As<Vector256<float>, Vector4>(ref currentBase), x);
+                ref Vector4 r = ref Unsafe.Add(ref Unsafe.As<Vector256<float>, Vector4>(ref resultBase), x);
+
+                if (p != c)
+                {
+                    r = c;
+
+                    // If is diff is found, the left side is marked by the min of previously found left side and the diff position.
+                    // The right is the max of the previously found right side and the diff position + 1.
+                    left = Math.Min(left, (int)x);
+                    right = Math.Max(right, (int)x + 1);
+                    hasRowDiff = true;
+                    hasDiff = true;
+                }
+                else
+                {
+                    r = replacement;
+                }
+            }
+
+            if (hasRowDiff)
+            {
+                if (top == int.MinValue)
+                {
+                    top = y;
+                }
+
+                bottom = y + 1;
+            }
+
+            PixelOperations<TPixel>.Instance.FromVector4Destructive(configuration, result, resultFrame.DangerousGetPixelRowMemory(y).Span, PixelConversionModifiers.Scale);
+        }
+
+        Rectangle bounds = Rectangle.FromLTRB(
+            left = Numerics.Clamp(left, 0, resultFrame.Width - 1),
+            top = Numerics.Clamp(top, 0, resultFrame.Height - 1),
+            Numerics.Clamp(right, left + 1, resultFrame.Width),
+            Numerics.Clamp(bottom, top + 1, resultFrame.Height));
+
+        // Webp requires even bounds
+        if (clampingMode == ClampingMode.Even)
+        {
+            bounds.Width = Math.Min(resultFrame.Width, bounds.Width + (bounds.X & 1));
+            bounds.Height = Math.Min(resultFrame.Height, bounds.Height + (bounds.Y & 1));
+            bounds.X = Math.Max(0, bounds.X - (bounds.X & 1));
+            bounds.Y = Math.Max(0, bounds.Y - (bounds.Y & 1));
+        }
+
+        return new(hasDiff, bounds);
+    }
+
+    public static void CopySource<TPixel>(ImageFrame<TPixel> source, ImageFrame<TPixel> destination, Rectangle bounds)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        Buffer2DRegion<TPixel> sourceBuffer = source.PixelBuffer.GetRegion(bounds);
+        Buffer2DRegion<TPixel> destBuffer = destination.PixelBuffer.GetRegion(bounds);
+        for (int y = 0; y < destBuffer.Height; y++)
+        {
+            Span<TPixel> sourceRow = sourceBuffer.DangerousGetRowSpan(y);
+            Span<TPixel> destRow = destBuffer.DangerousGetRowSpan(y);
+            sourceRow.CopyTo(destRow);
+        }
+    }
+}
+
+#pragma warning disable SA1201 // Elements should appear in the correct order
```

Comment:
Is this needed?

Reply:
Yeah, I wanted to keep the enum in the same file as the class, but Stylecop doesn't like it.